<launch>
  <arg name="uuv_name" />
  <arg name="start_time" default="-1"/>
  <arg name="radius" default="8"/>
  <arg name="center_x" default="0"/>
  <arg name="center_y" default="0"/>
  <arg name="center_z" default="-30"/>
  <arg name="n_points" default="50"/>
  <arg name="n_turns" default="1"/>
  <arg name="delta_z" default="5.0"/>
  <!-- Heading offset given in degrees -->
  <arg name="heading_offset" default="0"/>
  <arg name="duration" default="150"/>
  <arg name="max_forward_speed" default="0.3"/>

  <group ns="$(arg uuv_name)">
    <node pkg="uuv_control_utils"
      type="start_helical_trajectory.py"
      name="start_helical_trajectory"
      output="screen">
      <rosparam subst_value="true">
        start_time: $(arg start_time)
        radius: $(arg radius)
        center:
          - $(arg center_x)
          - $(arg center_y)
          - $(arg center_z)
        n_points: $(arg n_points)
        heading_offset: $(arg heading_offset)
        max_forward_speed: $(arg max_forward_speed)
        duration: $(arg duration)
        n_turns: $(arg n_turns)
        delta_z: $(arg delta_z)
      </rosparam>
    </node>
  </group>
</launch>
